hs_core/models.py

```diff
@@ -2119,12 +2161,16 @@ def check_irods_files(self, stop_on_error=False, log_errors=True,
                     errors.extend(error2)
                     ecount += ecount2
 
-            # Step 1: does every file here refer to an existing file in iRODS?
+            # Step 2: does every file here refer to an existing file in iRODS?
             for f in self.files.all():
                 if not istorage.exists(f.storage_path):
                     ecount += 1
                     msg = "check_irods_files: django file {} does not exist in iRODS"\
                         .format(f.storage_path)
+                    if clean_django:
+                        delete_resource_file(self.short_id, f.short_path, self.creator,
+                                             delete_logical_file=False)
```

Comment:
@alvacouch @pkdash I have seen some thread discussion on not deleting logical files when deleting the ResourceFile reference in Django, hence setting ```delete_logical_file``` to False here. I am wondering at what point should logical file is also deleted or it does not matter to leave the dangling logical file metadata around after all its physical file references are deleted?